src/cargo/command_bridge.rs

```diff
@@ -0,0 +1,141 @@
+//! This struct is a simple bridge that we can inspect for test purpose
+
+use std::collections::HashMap;
+use std::convert::{AsRef, Into};
+use std::default::Default;
+use std::ffi::{OsStr, OsString};
+use std::process::{Command, Stdio};
+
+/// This temporary struct is used to inspect the generated command before calling the process::Command
+/// builder, as we can't inspect its structure and make assertions on it
+#[derive(Debug, Default)]
+pub struct CommandBridge {
+    pub name: String,
+    pub args: Vec<OsString>,
+    pub env: HashMap<OsString, OsString>,
+    pub stdout: Option<Stdio>,
+    pub stderr: Option<Stdio>,
+}
+
+impl CommandBridge {
+    pub fn new(name: &str) -> Self {
+        CommandBridge {
+            name: String::from(name),
+            .. Default::default()
```

Comment:
This is such a neat trick!

Reply:
Ah ! Now that you say it, yes it is. Very useful to replace factories in tests: implement `Default` and only specify the field you want to test.

Reply:
I'll be leveraging this in some of my own stuff now!